src/helpers/jsonSchema.ts

```diff
@@ -0,0 +1,118 @@
+import type { JSONSchema7 } from "json-schema";
+
+import type { AnyObject } from "@/types/types";
+import { set } from "lodash";
+
+/**
+ * For a path like 'requests.1.request_type':
+ *
+ * obj = {
+ *   requests: [
+ *     { address: "", amount: "", request_type: "" },
+ *     { address: "", amount: "", request_type: "" }
+ *   ]
+ * }
+ *
+ * path = "requests.1.request_type"
+ * val = e.target.value
+ *
+ * This will update the value of requests[1].request_type
+ */
+const setDeepValue = (obj: AnyObject, path: string, val: any): AnyObject => {
+  if (!obj) return {};
+  const newObj = JSON.parse(JSON.stringify(obj));
```

Comment:
Should we use lossless JSON parse and stringify? Not sure if we're dealing with large numbers here.